packages/utils/src/severity.ts

```diff
@@ -1,20 +1,28 @@
-import { Severity } from '@sentry/types';
+import { Severity, SeverityLevel } from '@sentry/types';
 
-import { SeverityLevel, SeverityLevels } from './enums';
+// TODO: Should `severityFromString` be deprecated?
+
+// Note: Ideally the `SeverityLevel` type would be derived from `validSeverityLevels`, but that would mean either
```

Comment:
I kind of disagree that SeverityLevel should be derived from an array. Is this just so `severityFromString` is updated when we update the severities? I believe that just introduces headaches with dependencies.
I'd like for us to try to just keep Severity types from becoming anything runtime related.

I think we could get get rid of the the problems with a) and c) if we

- Delete `validSeverityLevels` here
- Add a type (not const!) in `@sentry/types` which looks like `export type ValidSeverityLevels = ['fatal', 'error', ...]`
- Change `SeverityLevel` in `@sentry/types` to `export type SeverityLevel = ValidSeverityLevels[number];`
- Define a local const in this file that looks like `const validSeverityLevels: ValidSeverityLevels = ['fatal', 'error', ...]` (We don't export that const)

That way we will throw a type error here if there is a mismatch in implementation (ie. if we forget to update something when changing severity levels).

We'd have to update the `severityFromString` function a tiny bit to make linters stop screaming but I think this would be more elegant than having a type derived from an actual const.

Reply:
> Edit: I just now realize that this comment is a hypothetical and a reminder lol.

Indeed. Also, everything you said, minus the last bullet point, is actually exactly how things are right now (prior to this PR). This note is partially inspired by that fact, like you said, as a reminder not to go back there.

Reply:
I believe right now we export a `const SeverityLevels` from the types package. I meant that I think we should convert that to `type SeverityLevels`. That should also probably also save some bytes now that I think of it 🤔

Reply:
Wherever it lives, and whatever we call it, it has to exist as an actual concrete array, for use in `SeverityLevelFromString`. So we don't get to save those bytes, unfortunately.